projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -465,3 +522,107 @@ def preview(
             formatting: Sets the basic report styling
         """
         self._save(self._preview_file.name, open=open, formatting=formatting)
+
+    ############################################################################
+    # Local served reports
+    def build(
+        self,
+        path: str,
+        formatting: t.Optional[ReportFormatting] = None,
+    ) -> None:
+        """Build a report which can be served by a local http server
+
+        Args:
+            path: File path to store the document
+            formatting: Sets the basic report styling
+        """
+
+        pl_path: Path = Path(path)
+        name = pl_path.stem[:127]
+        bundle_path = pl_path / SERVED_REPORT_BUNDLE_DIR
+        assets_path = pl_path / SERVED_REPORT_ASSETS_DIR
+
+        bundle_path.mkdir(parents=True, exist_ok=True)
+        assets_path.mkdir(parents=True, exist_ok=True)
+
+        self._served_local_writer.assert_bundle_exists()
+
+        # Copy across symlinked report bundle
+        copy_tree(str(self._served_local_writer.assets / "report"), str(bundle_path))
+
+        # Copy across symlinked Vue module
+        copyfile(str(self._served_local_writer.assets / VUE_ESM_FILE), str(pl_path / VUE_ESM_FILE))
+
+        local_doc, attachments = self._gen_report(embedded=False, served=True, title=name)
+
+        for a in attachments:
+            # TODO - compress in-memory to save a disk write?
+            # Copy across user report assets
+            with compress_file(a) as a_gz:
```

Comment:
Yeah this is a good point, @mands do you agree also?

Reply:
The only disadvantage is that local served reports would be considerably bigger, i.e. reports served through `Report.serve`. But maybe it's not a concern that the assets are large in dev mode. We could make the behaviour optional if it becomes a pain point for users

Reply:
a switch would be useful, as it could be useful for larger files.

We'll know more once we begin using it, and setting up our publishing tools down the road

Reply:
I'm happy for us to make it an option for now, maybe default to True, as serving locally will be the most common use-case.

As part of the new API we're considering as the follow-up, we'd probably have a separate `Processor` for uploading to GH Pages vs serving locally, so each processor would know whether to compress or not.